libraries/testing/include/eosio/testing/tester.hpp

```diff
@@ -147,7 +147,7 @@ namespace eosio::testing {
    struct produce_block_result_t {
       signed_block_ptr                   block;
       transaction_trace_ptr              onblock_trace;
-      std::vector<transaction_trace_ptr> traces;         // transaction traces
+      std::vector<transaction_trace_ptr> unapplied_transaction_traces; // unapplied from previous `push_block` not on head branch
```

Comment:
Undo this change. These are the transaction traces of the block. They are not:  `unapplied from previous push_block not no head branch`. Although, it is not really clear exactly what that is trying to convey. They are not unapplied transactions.

Reply:
Actually, these are the traces for the `unapplied_transactions` indeed. see [here](https://github.com/AntelopeIO/spring/blob/92a21962db0144c2174e1f66d308f72123382f08/libraries/testing/tester.cpp#L409-L417)

Reply:
traces from normal transaction within the block are returned by [push_transaction](https://github.com/AntelopeIO/spring/blob/92a21962db0144c2174e1f66d308f72123382f08/libraries/testing/include/eosio/testing/tester.hpp#L221)

Reply:
Those are traces returned from `push_transaction` https://github.com/AntelopeIO/spring/blob/92a21962db0144c2174e1f66d308f72123382f08/libraries/testing/tester.cpp#L410

Reply:
Oh, you are not capturing the traces of other `push_transaction` calls. This is not at all expected behavior. I think you have to go with the signal to capture all transaction traces.

Reply:
The unapplied transaction queue is going to be empty under most circumstances in the tests.

Reply:
yes, I also misunderstood what was happening here. I think with the renaming it is clear what it does, and replaces the other API `produce_block( std::vector<transaction_trace_ptr>& traces );` which the previous PR removed (it was used only in `forked_tests.cpp`). I agree it will be empty in most cases.

Reply:
I see now that it is only used in the forked_tests. Seems like the forked_tests could have done that via `applied_transaction` signal and not pollute the tester with this odd/confusing functionality.

Reply:
If you keep this then change the comment to something like: `// only traces of any unapplied transactions`
unapplied transactions can come from a fork (which is what the forked test is using this for) or aborted blocks.
My thought is this guy should have never added this confusing interface: https://github.com/EOSIO/eos/pull/8786

Reply:
I agree that the interface is really confusing (confused both of us), but I'll cut this guy some slack :-)
I'm happy to update `forked_tests` and remove this confusing functionality if you'd like.

Reply:
Seems best to remove it if not too much effort.

Reply:
Or I could add in the return value `scheduled_transactions_traces` to return traces of the scheduled transactions? Maybe these could be handy?

Reply:
Without a use-case (especially now that scheduled transactions are disabled) see no reason to add that.

Reply:
OK, I'll give the cleanup a try.

Reply:
These `unapplied_transactions` in libtester are also used when we produce an empty block [here](https://github.com/AntelopeIO/spring/blob/2f1b39f6b5e435e3c0f4b9c995fd88e1fc5a8b09/libraries/testing/include/eosio/testing/tester.hpp#L607-L610).

Which is weird because then we call `_produce_block` with `skip_pending_trxs = true`, which means we don't call `control->push_transaction()` on the `unapplied_transactions`.

I'm not quite understanding the purpose of this?

Reply:
Processing the unapplied transactions (default of `skip_pending_trxs = false`) mimics what the produce plugin does. Aborted blocks (an empty block aborts any transactions so that it is empty), and forked out blocks put there transactions in the unapplied transaction queue. The producer plugin always processed unapplied transaction as the first step of producing a block. The idea is that any forked out trxs or aborted block transactions are imediately attempted again in the next block before the end of `start_block` and other incoming transactions can be processed.

Reply:
OK, thanks, I understand a little bit better I think.

I think it is better to leave this PR  as-is. Also `forked_tests.cpp` verifies that transactions on fork are reported by `push_block` in order (and at the time the transactions have not been executed), so we do need this `unapplied_transaction_queue` for that purpose. I don't know that I can meaningfully simplify what we have now.

Reply:
Sounds good. Please do update the comment.

Reply:
Thanks. Comment is updated.